hypervisor/src/mshv/mod.rs

```diff
@@ -1123,29 +1053,39 @@ impl cpu::Vcpu for MshvVcpu {
                                             as usize;
                                     // Sanity check to make sure data len is within supported range.
                                     assert!(data_len <= 0x8);
-                                    let mut arg: mshv_read_write_gpa =
-                                        mshv_bindings::mshv_read_write_gpa {
-                                            base_gpa: src_gpa,
-                                            byte_count: data_len as u32,
-                                            ..Default::default()
-                                        };
 
-                                    self.fd
-                                        .gpa_read(&mut arg)
-                                        .map_err(|e| cpu::HypervisorCpuError::GpaRead(e.into()))?;
+                                    let mut data = vec![0; data_len];
+                                    self.gpa_read(src_gpa, &mut data)?;
 
                                     if let Some(vm_ops) = &self.vm_ops {
-                                        vm_ops
-                                            .mmio_write(dst_gpa, &arg.data[0..data_len])
-                                            .map_err(|e| {
-                                                cpu::HypervisorCpuError::RunVcpu(e.into())
-                                            })?;
+                                        vm_ops.mmio_write(dst_gpa, &data).map_err(|e| {
+                                            cpu::HypervisorCpuError::RunVcpu(e.into())
+                                        })?;
                                     }
 
                                     // Clear the SW_EXIT_INFO1 register to indicate no error
                                     self.clear_swexit_info1(ghcb_gpa)?;
                                 }
-                                SVM_EXITCODE_SNP_GUEST_REQUEST => {
+                                SVM_EXITCODE_SNP_GUEST_REQUEST
+                                | SVM_EXITCODE_SNP_EXTENDED_GUEST_REQUEST => {
+                                    if exit_code == SVM_EXITCODE_SNP_EXTENDED_GUEST_REQUEST {
+                                        warn!("Fetching extended guest request is not supported");
```

Comment:
I think this is unnecessary warn message since the feature is not supported, we can mark this as info instead. Because warn sometimes create a wrong impression that something is wrong with our stack.